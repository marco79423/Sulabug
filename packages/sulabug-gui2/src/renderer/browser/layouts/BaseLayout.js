import React, {useState} from 'react'
import classNames from 'classnames'
import {Link} from 'react-router-dom'
import {makeStyles, useTheme} from '@material-ui/core/styles'
import Drawer from '@material-ui/core/Drawer'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import List from '@material-ui/core/List'
import Divider from '@material-ui/core/Divider'
import IconButton from '@material-ui/core/IconButton'
import MenuIcon from '@material-ui/icons/Menu'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import ChevronRightIcon from '@material-ui/icons/ChevronRight'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import MoveToInboxIcon from '@material-ui/icons/MoveToInbox'
import ExploreIcon from '@material-ui/icons/Explore'
import SettingsIcon from '@material-ui/icons/Settings'
import CollectionsIcon from '@material-ui/icons/Collections'
import CssBaseline from '@material-ui/core/CssBaseline'

import 'typeface-roboto/index.css'

import SearchBar from '../components/SearchBar'


const drawerWidth = 240

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
    zIndex: 1,
    overflow: 'hidden',
    position: 'relative',
    display: 'flex',
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  SearchBarWrapper: {
    flex: 1,
    paddingRight: theme.spacing(2),
  },
  menuButton: {
    marginLeft: 12,
    marginRight: 36,
  },
  hide: {
    display: 'none',
  },
  drawerPaper: {
    position: 'relative',
    whiteSpace: 'nowrap',
    width: drawerWidth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerPaperClose: {
    overflowX: 'hidden',
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    width: theme.spacing(7),
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing(9),
    },
  },
  toolbar: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    ...theme.mixins.toolbar,
  },
  main: {
    flexGrow: 1,
    backgroundColor: theme.palette.background.default,

    display: 'flex',
    flexDirection: 'column',
    height: '100vh',
  },
  topContentSpacer: {
    ...theme.mixins.toolbar,
  },
  content: {
    height: '100%',
  }
}))

export default function BaseLayout({children}) {
  const theme = useTheme()
  const classes = useStyles(theme)
  const [isOpen, setOpen] = useState(false)

  const handleDrawerOpen = () => setOpen(true)
  const handleDrawerClose = () => setOpen(false)

  return (
    <React.Fragment>
      <CssBaseline/>
      <div className={classes.root}>
        <AppBar
          position='absolute'
          className={classNames(classes.appBar, isOpen && classes.appBarShift)}
        >
          <Toolbar disableGutters={!isOpen}>
            <IconButton
              color='inherit'
              aria-label='open drawer'
              onClick={handleDrawerOpen}
              className={classNames(classes.menuButton, isOpen && classes.hide)}
            >
              <MenuIcon/>
            </IconButton>
            <div className={classes.SearchBarWrapper}>
              <SearchBar/>
            </div>
          </Toolbar>
        </AppBar>
        <Drawer
          variant='permanent'
          classes={{
            paper: classNames(classes.drawerPaper, !isOpen && classes.drawerPaperClose),
          }}
          open={isOpen}
        >
          <div className={classes.toolbar}>
            <IconButton onClick={handleDrawerClose}>
              {theme.direction === 'rtl' ? <ChevronRightIcon/> : <ChevronLeftIcon/>}
            </IconButton>
          </div>
          <Divider/>
          <List>
            <ListItem button component={Link} to="/browser/collection">
              <ListItemIcon>
                <CollectionsIcon/>
              </ListItemIcon>
              <ListItemText primary="我的收藏"/>
            </ListItem>
            <ListItem button component={Link} to="/browser/browse">
              <ListItemIcon>
                <ExploreIcon/>
              </ListItemIcon>
              <ListItemText primary="瀏覽漫畫"/>
            </ListItem>
            <ListItem button component={Link} to="/browser/download">
              <ListItemIcon>
                <MoveToInboxIcon/>
              </ListItemIcon>
              <ListItemText primary="下載狀態"/>
            </ListItem>
          </List>
          <Divider/>
          <List>
            <ListItem button component={Link} to="/browser/settings">
              <ListItemIcon>
                <SettingsIcon/>
              </ListItemIcon>
              <ListItemText primary="個人設定"/>
            </ListItem>
          </List>
        </Drawer>
        <div className={classes.main}>
          <div className={classes.topContentSpacer}/>
          <div className={classes.content}>
            {children}
          </div>
        </div>
      </div>
    </React.Fragment>
  )
}
