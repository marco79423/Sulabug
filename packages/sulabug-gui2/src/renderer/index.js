import React from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter as Router, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'
import {Provider} from 'react-redux'

import configureStore from './configureStore'
import Browser from './browser'

const store = configureStore()

function App() {
  return (
    <Provider store={store}>
      <Router>
        <Switch>
          <Redirect exact from='/' to='/browser'/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Router>
    </Provider>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
