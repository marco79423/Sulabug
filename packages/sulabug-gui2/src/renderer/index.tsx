import React, {lazy, Suspense} from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'

const Manager = lazy(() => import('./manager'))
const Browser = lazy(() => import('./browser'))

function App() {
  return (
    <HashRouter>
      <Suspense fallback={<div>loading</div>}>
        <Switch>
          <Redirect exact from='/' to='/manager'/>
          <Route path="/manager" component={Manager}/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Suspense>
    </HashRouter>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
