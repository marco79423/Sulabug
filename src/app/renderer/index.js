import * as React from 'react'
import ReactDOM from 'react-dom'

import 'reflect-metadata'

import App from '../../imports/app/App'

ReactDOM.render(
  <App/>,
  document.getElementById('app')
)
