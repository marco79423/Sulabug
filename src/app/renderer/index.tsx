import * as React from 'react'
import {render} from 'react-dom'
import App from '../../imports/ui/pages/App'

render(
  <App/>,
  document.getElementById('app')
)