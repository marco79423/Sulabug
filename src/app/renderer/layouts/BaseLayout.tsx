import * as React from 'react'

import 'typeface-roboto/index.css'

interface PropsTypes {
}

export default class BaseLayout extends React.Component<PropsTypes, {}> {
  render() {
    return (
      <div>
        {this.props.children}
      </div>
    )
  }
}
