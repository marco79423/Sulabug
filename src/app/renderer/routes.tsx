import React from 'react'
import { Router, Route, Switch } from 'react-router'
import createBrowserHistory from 'history/createBrowserHistory'

import BrowsePage from '../../imports/ui/pages/App'

const browserHistory = createBrowserHistory()

export const renderRoutes = () => (
  <Router history={browserHistory}>
    <Switch>
      <Route exact path="/" component={BrowsePage}/>
    </Switch>
  </Router>
)
