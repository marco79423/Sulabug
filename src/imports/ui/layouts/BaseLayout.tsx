import * as React from 'react'
import classNames from 'classnames'
import {withStyles} from '@material-ui/core/styles'
import Drawer from '@material-ui/core/Drawer'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import List from '@material-ui/core/List'
import Divider from '@material-ui/core/Divider'
import IconButton from '@material-ui/core/IconButton'
import MenuIcon from '@material-ui/icons/Menu'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import ChevronRightIcon from '@material-ui/icons/ChevronRight'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import MoveToInboxIcon from '@material-ui/icons/MoveToInbox'
import ExploreIcon from '@material-ui/icons/Explore'
import SettingsIcon from '@material-ui/icons/Settings'
import CssBaseline from '@material-ui/core/CssBaseline'
import {Link} from 'react-router-dom'

import 'typeface-roboto/index.css'

import SearchBar from '../components/SearchBar'

interface PropsTypes {
  classes: any,
  theme: any,
}


const drawerWidth = 240

const styles = (theme: any) => ({
  root: {
    flexGrow: 1,
    zIndex: 1,
    overflow: 'hidden',
    position: 'relative',
    display: 'flex',
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  SearchBarWrapper: {
    flex: 1,
    paddingRight: theme.spacing.unit * 2,
  },
  menuButton: {
    marginLeft: 12,
    marginRight: 36,
  },
  hide: {
    display: 'none',
  },
  drawerPaper: {
    position: 'relative',
    whiteSpace: 'nowrap',
    width: drawerWidth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerPaperClose: {
    overflowX: 'hidden',
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    width: theme.spacing.unit * 7,
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing.unit * 9,
    },
  },
  toolbar: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    ...theme.mixins.toolbar,
  },
  main: {
    flexGrow: 1,
    backgroundColor: theme.palette.background.default,

    display: 'flex',
    flexDirection: 'column',
  },
  topContentSpacer: {
    ...theme.mixins.toolbar,
  },
  content: {
    flex: '1 300px',
    overflowY: 'auto',
    '&::-webkit-scrollbar': {
      display: 'none',
    },
  }
})


class BaseLayout extends React.Component<PropsTypes, {}> {
  state = {
    open: false,
  }

  handleDrawerOpen = () => {
    this.setState({open: true})
  }

  handleDrawerClose = () => {
    this.setState({open: false})
  }

  render() {
    const {classes, theme} = this.props

    return (
      <React.Fragment>
        <CssBaseline/>
        <div className={classes.root}>
          <AppBar
            position='absolute'
            className={classNames(classes.appBar, this.state.open && classes.appBarShift)}
          >
            <Toolbar disableGutters={!this.state.open}>
              <IconButton
                color='inherit'
                aria-label='open drawer'
                onClick={this.handleDrawerOpen}
                className={classNames(classes.menuButton, this.state.open && classes.hide)}
              >
                <MenuIcon/>
              </IconButton>
              <div className={classes.SearchBarWrapper}>
                <SearchBar/>
              </div>
            </Toolbar>
          </AppBar>
          <Drawer
            variant='permanent'
            classes={{
              paper: classNames(classes.drawerPaper, !this.state.open && classes.drawerPaperClose),
            }}
            open={this.state.open}
          >
            <div className={classes.toolbar}>
              <IconButton onClick={this.handleDrawerClose}>
                {theme.direction === 'rtl' ? <ChevronRightIcon/> : <ChevronLeftIcon/>}
              </IconButton>
            </div>
            <Divider/>
            <List>
              <ListItem button component={Link} to="/">
                <ListItemIcon>
                  <ExploreIcon/>
                </ListItemIcon>
                <ListItemText primary="瀏覽漫畫"/>
              </ListItem>
              <ListItem button component={Link} to="/download">
                <ListItemIcon>
                  <MoveToInboxIcon/>
                </ListItemIcon>
                <ListItemText primary="我的下載"/>
              </ListItem>
            </List>
            <Divider/>
            <List>
              <ListItem button component={Link} to="/settings">
                <ListItemIcon>
                  <SettingsIcon/>
                </ListItemIcon>
                <ListItemText primary="個人設定"/>
              </ListItem>
            </List>
          </Drawer>
          <div className={classes.main}>
            <div className={classes.topContentSpacer}/>
            <div className={classes.content}>
              {this.props.children}
            </div>
          </div>
        </div>
      </React.Fragment>
    )
  }
}


export default withStyles(styles, {withTheme: true})(BaseLayout)