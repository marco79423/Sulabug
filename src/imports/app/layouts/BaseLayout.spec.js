import 'reflect-metadata'

import React from 'react'
import renderer from 'react-test-renderer'
import BaseLayout from './BaseLayout'


describe('BaseLayout', () => {
  it('will render layout correctly', async () => {
    const component = renderer.create(
      <BaseLayout
        searchComic={() => {}}
        changeCurrentPage={() => {}}
      >
        hello world
      </BaseLayout>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
