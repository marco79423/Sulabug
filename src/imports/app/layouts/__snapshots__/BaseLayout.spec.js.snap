// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout will render layout correctly 1`] = `
<div
  className="BaseLayout-root-1"
>
  <header
    className="MuiPaper-root-22 MuiPaper-elevation4-28 MuiAppBar-root-13 MuiAppBar-positionAbsolute-15 MuiAppBar-colorPrimary-20 BaseLayout-appBar-2"
  >
    <div
      className="MuiToolbar-root-49 MuiToolbar-regular-51"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root-59 MuiIconButton-root-53 MuiIconButton-colorInherit-54 BaseLayout-menuButton-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-58"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-62"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-207"
        />
      </button>
      <div
        className="BaseLayout-SearchBarWrapper-4"
      >
        <div
          className="SearchBar-root-71"
        >
          <div
            className="MuiInputBase-root-88 MuiInput-root-75 SearchBar-input-73"
            onClick={[Function]}
          >
            <input
              aria-label="Description"
              autoFocus={true}
              className="MuiInputBase-input-98 MuiInput-input-83"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="請輸入想試看的漫畫……"
              type="text"
            />
          </div>
          <button
            className="MuiButtonBase-root-59 MuiButton-root-105 MuiButton-contained-116 MuiButton-containedSecondary-118 MuiButton-raised-119 MuiButton-raisedSecondary-121 MuiButton-sizeSmall-128 SearchBar-button-74"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-106"
            >
              搜尋
            </span>
            <span
              className="MuiTouchRipple-root-207"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    className="MuiDrawer-root-131 MuiDrawer-docked-132"
  >
    <div
      className="MuiPaper-root-22 MuiPaper-elevation0-24 MuiDrawer-paper-133 BaseLayout-drawerPaper-7 BaseLayout-drawerPaperClose-8 MuiDrawer-paperAnchorLeft-134 MuiDrawer-paperAnchorDockedLeft-138"
    >
      <div
        className="BaseLayout-toolbar-9"
      >
        <button
          className="MuiButtonBase-root-59 MuiIconButton-root-53"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-58"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-62"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-207"
          />
        </button>
      </div>
      <hr
        className="MuiDivider-root-143"
      />
      <ul
        className="MuiList-root-148 MuiList-padding-149"
      >
        <a
          className="MuiButtonBase-root-59 MuiListItem-root-152 MuiListItem-default-155 MuiListItem-gutters-160 MuiListItem-button-161"
          href="/main/browse"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemIcon-root-164"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-62"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 10.9c-.61 0-1.1.49-1.1 1.1s.49 1.1 1.1 1.1c.61 0 1.1-.49 1.1-1.1s-.49-1.1-1.1-1.1zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm2.19 12.19L6 18l3.81-8.19L18 6l-3.81 8.19z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-165"
          >
            <span
              className="MuiTypography-root-171 MuiTypography-subheading-178 MuiListItemText-primary-168"
            >
              瀏覽漫畫
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-207"
          />
        </a>
        <a
          className="MuiButtonBase-root-59 MuiListItem-root-152 MuiListItem-default-155 MuiListItem-gutters-160 MuiListItem-button-161"
          href="/main/collection"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemIcon-root-164"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-62"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4l2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-165"
          >
            <span
              className="MuiTypography-root-171 MuiTypography-subheading-178 MuiListItemText-primary-168"
            >
              我的收藏
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-207"
          />
        </a>
        <a
          className="MuiButtonBase-root-59 MuiListItem-root-152 MuiListItem-default-155 MuiListItem-gutters-160 MuiListItem-button-161"
          href="/main/download"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemIcon-root-164"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-62"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-165"
          >
            <span
              className="MuiTypography-root-171 MuiTypography-subheading-178 MuiListItemText-primary-168"
            >
              下載狀態
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-207"
          />
        </a>
      </ul>
      <hr
        className="MuiDivider-root-143"
      />
      <ul
        className="MuiList-root-148 MuiList-padding-149"
      >
        <a
          className="MuiButtonBase-root-59 MuiListItem-root-152 MuiListItem-default-155 MuiListItem-gutters-160 MuiListItem-button-161"
          href="/main/settings"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemIcon-root-164"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-62"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h20v20H0V0z"
                fill="none"
                transform="scale(1.2, 1.2)"
              />
              <path
                d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                transform="scale(1.2, 1.2)"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-165"
          >
            <span
              className="MuiTypography-root-171 MuiTypography-subheading-178 MuiListItemText-primary-168"
            >
              個人設定
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-207"
          />
        </a>
      </ul>
    </div>
  </div>
  <div
    className="BaseLayout-main-10"
  >
    <div
      className="BaseLayout-topContentSpacer-11"
    />
    <div
      className="BaseLayout-content-12"
    >
      hello world
    </div>
  </div>
</div>
`;
